### DISCUSSION TITLE
Rename 'master' branch

### DISCUSSION THREAD

**Discussion by matthias-ronge** [kitodo/kitodo-production#4533] [General]
It is a trend in Git to rename all “master” branches to something else, usually “main”. The background is that, in English, the _master_ word has the connotation of a lord, in the sense of slave driver. Personally, as a German speaker, I don’t read it that way, for me it's more like _masterly_, and that's probably what it was meant to be. But political correctness always means: not saying things that someone who is stupid, or who deliberately tries to misunderstand the other, might say, it was meant that way. So, if we do this, we will be paying tribute to many former slaves in the English-speaking part of the world or their descendants, so I would like to suggest such a change.

For all co-developers in Git, you might have to adjust the configuration a little.

**Comment by stweil** [comment]
I support this renaming (and did it already for other repositories, for example all repositories of https://github.com/tesseract-ocr/).

The rename process is easy, but can only be done by a maintainer. And of course it should be announced before it is done. When would be a good time for it? Maybe now?

**Comment by aetherfaerber** [comment]
As I missed the opportunity to ask for an action on this issue I would like to push this discussion. @solth can you tell what would be needed to set this in motion?

**Reply by stweil** [reply]
@solth, I think this is something which only affects the developers. What about sending an e-mail on the mailing list and asking for feedback here? Then this could be done very fast.

**Reply by solth** [reply]
Technically this is no problem. I would think it would be good if at least the @kitodo/kitodo-board would give this a "go", though, and to coordinate this change with Kitodo.Presentation, which still uses the same default branch names as production (Kitodo.Publication on the other hand has already renamed it's `master` branch to `main`).

**Reply by stweil** [reply]
I don't think that a board decision is needed, but as you already asked, an answer for all Kitodo repositories would of course be helpful.

Technically there is no need that the actual change from master to main occurs at the same time for all repositories, so Kitodo.Production could start without waiting for Kitodo.Presentation (which might want to coordinate with slub/dfgviewer and slub/slub_digitalcollections).

**Reply by stweil** [reply]
Cc @beatrycze-volk and @sebastian-meyer to notice them about this ongoing discussion which concerns Kitodo.Presentation, too.

**Comment by sebastian-meyer** [comment]
I totally missed this discussion. Thanks for keeping me in the loop, @stweil!

I am fine with renaming the `master` branch to `main`.

**Comment by stefanCCS** [comment]
Just one comment from me as member of @kitodo/kitodo-board:
I am fine with this, as long as our Release Managers are also fine with (as it looks like, both have already confirmed this).

**Comment by stweil** [comment]
@solth, thanks for the new releases and the good wishes. What about ending the year now with a new main branch?

**Reply by sebastian-meyer** [reply]
I just renamed the main branch for Kitodo.Presentation as suggested. In order to rename your local branch as well and still track the remote branch you can use the following commands:

```
git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
```

**Reply by stweil** [reply]
... and don't forget to rename the master branch in personal forks, too.

**Reply by solth** [reply]
I followed suit and renamed the `master` branch to `main` in Kitodo.Production as well.

**Reply by henning-gerhardt** [reply]
You should not forget to adjust all workflow files if they used `master` for executing, comparing, ... if not they did not get executed or do stuff in a wrong way.

**Reply by solth** [reply]
> You should not forget to adjust all workflow files if they used `master` for executing, comparing, ... if not they did not get executed or do stuff in a wrong way.

Thanks a lot for pointing this out! I will open a pull request to adjust the workflow files. Perhaps you could then check whether I forgot something?

**Comment by solth** [comment]
Marking discussion as "closed" as `master` branch has been renamed to `main`.
